### PR TITLE
Changes to schema to loosen dependencies involving Forced ID table

### PR DESCRIPTION
**Pull request by IanMMarshall**
This is a proposed and potentially breaking change intended to improve performance and should not be automatically merged.

Proposed change involves removing forced_id_pid columns from Resource and Resource History tables in the JPA server model and changing how the ResourceTable and ResourceHistory entities interact with each other and with the ForcedId entity.

### CODE REVIEW

**Comment by jamesagnew** [comment]
FYI- This looks great.

I'd like to leave this unmerged until after the release of 4.2.0 (feb 13) but then proceed right away afterward, in order to give us lots of time to identify regressions.

File: hapi-fhir-jpaserver-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java [hunk 1]
```diff
@@ -61,6 +61,11 @@ public HapiFhirJpaMigrationTasks(Set<String> theFlags) {
 
 	protected void init420() { // 20191015 - present
 		Builder version = forVersion(VersionEnum.V4_2_0);
+
+		// Eliminate circular dependency.
+		version.onTable("HFJ_RESOURCE").dropColumn("20200130.1", "FORCED_ID_PID");
+		version.onTable("HFJ_RES_VER").dropColumn("20200130.2", "FORCED_ID_PID");
+		version.onTable("HFJ_RES_VER").addForeignKey("20200130.3", "FK_RESOURCE_HISTORY_RESOURCE").toColumn("RES_ID").references("HFJ_RESOURCE", "RES_ID");
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java]: Can you merge master into this branch, and adjust this to put these tasks in a new `init430` method with the current date on the tasks?

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/ExpungeEverythingService.java [hunk 2]
```diff
@@ -78,8 +78,6 @@ void expungeEverything(RequestDetails theRequest) {
 		ourLog.info("BEGINNING GLOBAL $expunge");
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/ExpungeEverythingService.java]: Overall comment- This could use a changelog.

Comment (IanMMarshall) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/ExpungeEverythingService.java]: New changelog added.

File: hapi-fhir-jpaserver-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java (hunk 1 above)
Comment (IanMMarshall) [reply hapi-fhir-jpaserver-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java]: Done.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/ExpungeEverythingService.java (hunk 2 above)
Comment (IanMMarshall) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/expunge/ExpungeEverythingService.java]: New changelog added.

**Pull request merged**